### ISSUE TITLE
Reopening a discussion on representing user accounts

### ISSUE THREAD

**Issue by rrlevering** [schemaorg/schemaorg#3403]
At Google we are starting to release more guidelines for conveying creator/author/person information but the lack of a type for account makes the markup awkward.

There has been previous discussion for something like a user account or social profile account or whatever we want to scope it (see https://github.com/schemaorg/schemaorg/issues/267).  There is no representation AFAIK for something like that in schema.org.  Representationally this leads to problems figuring out how to represent domain-specific information that is separate from the real-world person that the account/profile represents.  Some examples:

- How do we represent when a profile was created?  That's not when the person was created.  At Google we are recommending hanging that on a ProfilePage node (so the page implicitly is created when the account is created) but that is not ideal.
- Naming is confusing distinguishing between a system handle and a real world name.  There is little specification for whether name (or alternateName) represents the canonical representation of the thing in its system context or something more objective.
- Ambiguity in entity type.  Some systems cannot tell whether something is a Person or Organization and yet many of our author properties have to use a specific type to be valid.

**Comment by WeaverStever** [comment]
I wrote (seems to be gone now) about creating a specification where creatives could publish their various [taxonomy](https://en.wikipedia.org/wiki/Folksonomy) handles and hashtags (content / discussion threads) on their website schema.  

In the structured data, the handle or hashtag would be need to be paired with the social media domain name, creating a site-search from there would be pretty simple.

~~~~
{
  "name" : "@parTEE",
  "domain": "jagermeister.com",
  "discussionThreads": ["#allNightLong", "#sloHungover", "#holdMyBeer"]
}
~~~~

I guess the publisher could also declare if they are a person, organization, musicalgroup etc., as this would help with the disambiguation.

**Comment by rrlevering** [comment]
The structure I'm seeing in my head looks like:

```
{
   "@type": "SocialMediaPosting",
   "author": {
        "@type": "UserAccount",
        "url": "http://example.com/profiles/123"
        ... whatever fields are present in the content (also potentially a real-world object)
   }
   ...
}
```

And on http://example.com/profiles/123:

```
{
   "@type": "ProfilePage,
   "mainEntityOfPage": {
       "@type": "UserAccount",
       "name": "@jennlovesdogs",
       "dateCreated": "20231223T12:30Z",
       "accountSystem": "example.com", // Though we would totally infer this in reality
       "accountOwner": {
            "@type": "Person",
            "name": "Jennifer Franklin"
      }
}
```

So you've pulled apart the concept of the domain account from the real world entity.  I guess it's a bit more verbose because there's a nested object now whereas you are collapsing, but I think a lot of sites really just use the account concept on the content page (and might not even have a "real-world" concept if it's more of an anonymous thing) so it's not any more verbose.

**Comment by MatthiasWiesmann** [comment]
I suspect the `accountOwner` would be either a `Person` or an `Organization`?

**Comment by rrlevering** [comment]
Yep.  So it sort of has the same range on the actual real-life entity, just with an in-between account object.

**Comment by shaedrich** [comment]
I see the need for this entity in an online context, however, would it be better or an additional option, to have an `Identity` entity, that is a little broader?

**Comment by WeaverStever** [comment]
I ran across an interesting concept from the implementation of ads.txt  1.1 implementation.

My thinking here is that the social-profile will probably be one of many social-profiles and that the owner of the social-profile may want to monetize her published content regardless of the platform hosting the profile. Alternately, the profile owner may want to limit information that is exposed to advertisers. 

In the ads.txt implementation, the ads.txt file is paired with a seller.json (in adsense I believe) are properties like ownerdomain, managerdomian etc.  The ads.txt file allows the ownerdomain to whitelist advertising partners.  Various settings in sellers.json indicate that the ownerdomain is selling direct or uses a reseller. 

Anyway, it seems like there is an emerging ontology that might be consulted for similarity with the goals of this thread. (Instead of a website or domain, perhaps there could be a dashboard where the social-profile owner would participate.)

https://iabtechlab.com/wp-content/uploads/2022/04/Ads.txt-1.1-Implementation-Guide.pdf

**Comment by shaedrich** [comment]
> I see the need for this entity in an online context, however, would it be better or an additional option, to have an `Identity` entity, that is a little broader?

`Identity` can include social media accounts but also email addresses, maybe even personal websites 🤔

**Comment by WeaverStever** [comment]
I think I may have found an area that this topic fits into.
There is a [contactPoint ](https://schema.org/ContactPoint)type available to persons and organizations.  


[ContactPointOption](https://schema.org/ContactPointOption) has two enumerations

Enumeration members

- [HearingImpairedSupported](https://schema.org/HearingImpairedSupported)
- [TollFree](https://schema.org/TollFree)


I suggest adding a third enumeration

- socialMedia

~~~
<script type="application/ld+json">
{ "@context" : "https://schema.org",
  "@type" : "Organization",
  "url" : "http://www.t-mobile.com",
  "contactPoint" : [
    { "@type" : "ContactPoint",
      "telephone" : "+1-877-746-0909",
      "contactType" : "customer service",
      "contactOption" : "TollFree",
      "areaServed" : "US"
    } , {
      "@type" : "ContactPoint",
      "telephone" : "+1-505-998-3793",
      "contactType" : "customer service"
    } , {
      "@type" : "ContactPoint",
      "alternateName" : "@TMobile",
      "contactType" : "Twitter account",
      "contactOption" : ["SocialMedia"],
      "url": "https://x.com/tmobile",
      "potentialAction": {
        "startTime": "1999-12-01"     
      }
    }
  ] 
}
</script>
~~~

**Comment by WeaverStever** [comment]
Long story short add "socialMedia" to [ContactPointOption](https://schema.org/ContactPointOption) as an enumeration.

1. Some people use social media as a communications system.
2. Social media links do not really fit into the sameAs property.

**Comment by rrlevering** [comment]
I see some merit in what your saying, but:
a) The current ContactPoint doesn't fit a lot of the things I have in mind.  For instance, your potentialAction suggestion doesn't really make a lot of sense to me (what type of action is that supposed to be and it's not "potential")?
b) ContactPoint seems like its conflating two concepts.  One is a method of contact (email vs. phone) and the other is a channel/suborganization of communication (customer service vs. hr).  You see to be focused on the former but I personally read it more like the latter which fits less.
c) I don't think I see a social media account as mostly being a contact mechanism.  I actually see it more as a publishing mechanism which doesn't fit the same semantic model/set of properties.
d) I was hoping this concept could go beyond social media and be a model for any account object (particularly computing related) that was independent of a person (a person could have multiple accounts).  Once you get further away from social media (like an account on a network system), contact point makes even less sense.

**Comment by shaedrich** [comment]
> This issue is being nudged due to inactivity.

![](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExdzRrOHI2OW4zam1zZ3JhdmYwemZreTZmY3lnc3ltMWI5dnlqMGszYyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/6lScd4x2D5Oko/giphy.gif)

**Comment by WeaverStever** [comment]
@rrlevering 

> a) The current ContactPoint doesn't fit a lot of the things I have in mind. For instance, your potentialAction suggestion doesn't really make a lot of sense to me (what type of action is that supposed to be and it's not "potential")?

I included the potentialAction:startTime because of the ask from the original poster (Google) on this thread...  "How do we represent when a profile was created?"  startDate would be better, but potentialAction does not include the property. 

>  b) ContactPoint seems like its conflating two concepts. One is a method of contact (email vs. phone) and the other is a channel/suborganization of communication (customer service vs. hr). You see to be focused on the former but I personally read it more like the latter which fits less.

Yes, the profilePage  and socialMedia profile is already conflated.  A musician profile on Facebook is technically a profilePage about the artist, but is is also a critical method for communications. 

> c) I don't think I see a social media account as mostly being a contact mechanism. I actually see it more as a publishing mechanism which doesn't fit the same semantic model/set of properties.

I see the social media account as a tool for somebody who wants to have a public persona and a managed method of communications.  For instance, Taylor Swift's website does not list an email address, only social media sites.  Additionally, social media discussions are public facing, the chances of getting a response from her staff is much greater than an email to webmaster@taylorswift.com    

>  d) I was hoping this concept could go beyond social media and be a model for any account object (particularly computing related) that was independent of a person (a person could have multiple accounts). Once you get further away from social media (like an account on a network system), contact point makes even less sense.

Interesting, the current properties in contactPoint seem to have a few things  in common with LDAP. 

.

**Comment by WeaverStever** [comment]
@rrlevering 

Another point that should be included here is [folksonomy](https://en.wikipedia.org/wiki/Folksonomy) these ids and hashtag discussions are used liberally on social media platforms.  One of my artists blew up on TikTok with a rendition of the National Anthem, millions of views, yet I don't have a method to credit her real world persona as the creator of that content.

**Comment by shaedrich** [comment]
I'd view it more like this kind of hierarchy:
```mermaid
---
config:
    treemap:
        showValues: false
---
treemap-beta
"Real person"
    "Persona (on- and offline)"
        "contact method": 1
        "online presence": 1
```

**Comment by WeaverStever** [comment]
@shaedrich 

I agree, I don't think this can be shoehorned into a profilePage because they tend to be understood as domain specific. contactPoint already has most of the needed properties.

contactPoint on a profilePage might work nicely.

**Comment by WeaverStever** [comment]
Related?

[Thing](https://schema.org/Thing) > [Action](https://schema.org/Action) > [InteractAction](https://schema.org/InteractAction)

More specific Types
[BefriendAction](https://schema.org/BefriendAction)
[CommunicateAction](https://schema.org/CommunicateAction)
[FollowAction](https://schema.org/FollowAction)
[JoinAction](https://schema.org/JoinAction)
[LeaveAction](https://schema.org/LeaveAction)
[MarryAction](https://schema.org/MarryAction)
[RegisterAction](https://schema.org/RegisterAction)
[SubscribeAction](https://schema.org/SubscribeAction)
[UnRegisterAction](https://schema.org/UnRegisterAction)

**Comment by MatthiasWiesmann** [comment]
I would argue that people usually mary [Person](https://schema.org/Person), a UserAccount would generally be a proxy for a person, not the person itself.
